components/style/bloom.rs

```diff
@@ -216,9 +216,11 @@ impl StyleBloom {
         // until we find a common ancestor.
         while *self.elements.last().unwrap() != common_parent.as_node().to_unsafe() {
             parents_to_insert.push(common_parent);
-            common_parent =
-                common_parent.parent_element().expect("We were lied again?");
             self.pop::<E>().unwrap();
+            common_parent = match common_parent.parent_element() {
+                Some(parent) => parent,
+                None => break,
```

Comment:
This can leave stale parents in the filter, right?

I think for the AC you're talking about this doesn't happen, but probably we should either `assert!(self.elements.is_empty())`, `while self.pop::<E>().is_some() {}` or `return self.rebuild(element)`.

Also, please keep the panic in the servo case if you don't mind, since that way it's easier to catch errors, and add a comment on why we allow this, since it's not obvious reading the code :).